src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -663,13 +665,14 @@ private static String findSentenceEndingOrNull(String text, String period) {
             if (!DEFAULT_PERIOD.equals(period)
                 || afterPeriodIndex >= text.length()
                 || Character.isWhitespace(text.charAt(afterPeriodIndex))) {
-                sentenceEnding = text.substring(0, periodIndex);
+                final String resultStr = text.substring(0, periodIndex);
+                result = Optional.of(resultStr);
                 break;
             }
             else {
                 periodIndex = text.indexOf(period, afterPeriodIndex);
             }
         }
-        return sentenceEnding;
+        return result;
```

Comment:
@mohitsatr what are the tradeoffs between using Optionals throughout the method as you've done here, or returning `Optional.ofNullable(result)` here instead?

Reply:
allow me to explain how this logic is working: 

In `findSentenceEnding` function, `result` can have two possible values: an Empty-optional or an optional containing a String value.

In `getFirstSentence` func, we do :
`sentenceEnding = findSentenceEnding(test, period);`

if `sentenceEnding.isPresent()` is true (the case when we have successfully found the sentence ending) : 
it means Optional contains a String value (empty string "" is also valid). it is when we break the loop.

if `isPresent()` is false: it is empty-optional. ending is not found yet. keep looping.

I was not aware of `.ofNullable` api. After looking into it, i think **not using it** makes more sense in this case because `Optional.ofNullable(result)` would only make sense in null based logic.(i.e when `result` can be either a string or a null as in original code). Also isn't the sole purpose of all this refactoring is to eliminate null from the code ?

Reply:
The point of this issue is to not **return** `null` from these private methods - it is not desirable (or possible) to avoid null completely in Java :). We have no issue with usage of `null` inside of these methods. My point is why make the method body more complicated when we can keep everything the same and just return an optional.

Reply:
I see no problem to have Optional in this special case, if such code was written from the beginning I would not ask to make in on nulls.
As we started doing de-nullozation in this Check, for some reason, I was completely ok to do a bit more refactoring to try to avoid null. Code is kind of same before and after refactoring by ease of understanding, just more fashional on Optional.

Reply:
Using optional has nothing to do with “fashion”, it is an API that should be used judiciously and deliberately like any other. It is not designed to replace all null usage. The changes in this PR are a clear abuse of the API, and violate Oracle’s own advice about Optionals usage (see link below).

Reply:
I agree with @nrmancuso , I don't understand why we avoid `Optional.ofNullable`?

Reply:
please consider https://github.com/checkstyle/checkstyle/pull/16020/files#r1893954538

Reply:
Thanks, I got it. But I have a wider question: why can't we use an empty string to represent an absent sentence ending? I tried to do this in this experimental PR: https://github.com/checkstyle/checkstyle/pull/16040

Reply:
hi @rdiachenko, this was the first approach i came up with  here: https://github.com/checkstyle/checkstyle/pull/15965#pullrequestreview-2466226778 as it is simple and avoid null and optionals altogether.

Reply:
Looks like all others likes to read and empty first sentence is very rare, so we can mix actual empty(with period )and no content like `/***/`.
For now we can accept this, if users complain, we will improve.

Reply:
I am 100% on board with returning Optional from this method; it makes sense because the sentence is actually optional. I do not agree with the usage throughout the method body in some of these methods. 

Like I said, I won’t block this PR, but I am not a fan of this new fascination with “fashionable” code. Someone once told me that good, easy to read code is a result of hard writing; Optional should not be used as a crutch.

To be honest, we are bikeshedding here and wasting maintainer and contributor time. I would like to see @mohitsatr devote their time to something more interesting, valuable, and rewarding.